vault/token_store.go

```diff
@@ -1162,26 +1225,46 @@ func (ts *TokenStore) revokeSalted(ctx context.Context, saltedID string) (ret er
 
 	// Clear the secondary index if any
 	if entry.Parent != "" {
-		parentSaltedID, err := ts.SaltID(ctx, entry.Parent)
-		if err != nil {
-			return err
+		var parentIndex string
```

Comment:
This code makes an invalid assumption that the parent will be the same version as the child.

Reply:
No it doesn't. It only assumes that tokens with version 2 will always have their parent indexes HMACed, regardless of the parent's version, which seems like the right thing to do.